packages/maskbook/src/plugins/Trader/trader/ether/useTrade.ts

```diff
@@ -0,0 +1,25 @@
+import BigNumber from 'bignumber.js'
+import { useAsyncRetry } from 'react-use'
+import { CONSTANTS } from '../../../../web3/constants'
+import { isSameAddress } from '../../../../web3/helpers'
+import { useConstant } from '../../../../web3/hooks/useConstant'
+import { ERC20TokenDetailed, EthereumTokenType, EtherTokenDetailed } from '../../../../web3/types'
+
+const ZERO = new BigNumber(0)
+
+export function useTrade(
+    inputToken?: EtherTokenDetailed | ERC20TokenDetailed,
+    outputToken?: EtherTokenDetailed | ERC20TokenDetailed,
+) {
+    const WETH_ADDRESS = useConstant(CONSTANTS, 'WETH_ADDRESS')
+
+    return useAsyncRetry(async () => {
```

Comment:
why need async?

Reply:
To keep the same interface as a trade provider.